### ISSUE TITLE
RFC: Unicode Support

### ISSUE THREAD

**Issue by Qix-** [arua-lang/proposal#11]
_~TODO~_

**Comment by chrmatique** [comment]
As a question, I'm assuming when you say unicode support you mean UTF-8? As there are several UTF formats, UTF-8 being the most common.

**Comment by Qix-** [comment]
UTF-8, UTF-16, UTF-32 and UTF-64.

``` arua
typedef [u8] as str8
typedef [u16] as str16
typedef [u32] as str32
typedef [u64] as str64

alias str8 as str
```

**Comment by chrmatique** [comment]
@Qix- 
Sweet. That certainly will be nice, having full UTF8-64 support.

**Comment by Qix-** [comment]
@Polygn yep see the updated #3 for why the `typedef` and `alias` keywords are how they are in the above comment.

**Comment by chrmatique** [comment]
@Qix- ahhh I see. So `alias` "renames" but without the copy of the original type, `typedef` copies the original type and renames. I quite like that because using `alias` could save bit space where needed.

**Comment by Qix-** [comment]
`typedef` creates a new type out of an existing type. `alias` just creates an alias for a type.

You can think of it like this:

``` arua
typedef Foo as Bar
```

``` c++
class Bar : private Foo {};
```

and

``` arua
alias Foo as Bar
```

``` c++
#define Bar Foo
```
